src/test/app/NFToken_test.cpp

```diff
@@ -6798,6 +6798,320 @@ class NFToken0_test : public beast::unit_test::suite
+                env.close();
+
+                // after prev transaction, Bob owns 199,999,990 drops due to
+                // burnt tx fee
+
+                BEAST_EXPECT(ownerCount(env, bob) == 0);
+
+                // Send bob an increment reserve and 10 drops (to make up for
+                // the transaction fee burnt from the prev failed tx) Bob now
+                // owns 250,000,000 drops
+                env(pay(env.master, bob, incReserve + drops(10)));
+                env.close();
+
+                // However, this transaction will still fail because the reserve
+                // requirement is 10 drops higher
+                env(token::acceptSellOffer(bob, sellOfferIndex),
+                    ter(tecINSUFFICIENT_RESERVE));
+                env.close();
+
+                // Send bob an increment reserve and 20 drops
+                // Bob now owns 250,000,010 drops
+                env(pay(env.master, bob, drops(20)));
+                env.close();
+
+                // Bob is now able to accept the offer
+                env(token::acceptSellOffer(bob, sellOfferIndex));
+                env.close();
+
+                BEAST_EXPECT(ownerCount(env, bob) == 1);
+            }
+        }
+
+        // Now exercise the scenario when the buyer accepts
+        // many sell offers
+        {
+            Account const alice{"alice"};
+            Account const bob{"bob"};
+
+            Env env{*this, features};
+            auto const acctReserve = env.current()->fees().accountReserve(0);
+            auto const incReserve = env.current()->fees().increment;
+
+            env.fund(XRP(10000), alice);
+            env.close();
+
+            env.fund(acctReserve + XRP(1), bob);
+            env.close();
+
+            if (!features[fixNFTokenReserve])
+            {
+                // Bob can accept many NFTs without having a single reserve!
+                for (size_t i = 0; i < 200; i++)
+                {
+                    // alice mints an NFT and creates a sell offer for 0 XRP
+                    auto const sellOfferIndex =
+                        mintAndCreateSellOffer(env, alice, XRP(0));
+
+                    // Bob is able to accept the offer
+                    env(token::acceptSellOffer(bob, sellOfferIndex));
+                    env.close();
+                }
+            }
+            else
+            {
+                // alice mints the first NFT and creates a sell offer for 0 XRP
+                auto const sellOfferIndex1 =
+                    mintAndCreateSellOffer(env, alice, XRP(0));
+
+                // Bob cannot accept this offer because he doesn't have the
+                // reserve for the NFT
+                env(token::acceptSellOffer(bob, sellOfferIndex1),
+                    ter(tecINSUFFICIENT_RESERVE));
+                env.close();
+
+                // Give bob enough reserve
+                env(pay(env.master, bob, drops(incReserve)));
```

Comment:
shouldn't Bob have at least `incReserve + drops(10) /* current transaction fee */ + drops(10) /* fees lost due to the previous transaction*/` ?

I don't understand why this is different from testcase in line 6888 ?

Reply:
On that note, I found it easier to understand a simpler test case:
```
// Bob has enough reserve for the Transaction Fees (10 drops) and the reserve for a new NFTokenPage
env(pay(env.master, bob, incReserve + drops(10)));
env.close();
env(token::acceptSellOffer(bob, sellOfferIndex));
env.close();
 
BEAST_EXPECT(ownerCount(env, bob) == 1);
```

Reply:
> I don't understand why this is different from testcase in line 6888 ?

The test case in line 68888 is testing the precise reserve requirements in the unit of drops. While in this test, it is exercising the behavior when Bob accepts many NFTs.

> shouldn't Bob have at least incReserve + drops(10) /* current transaction fee */ + drops(10) /* fees lost due to the previous transaction*/ ?

In this test case, Bob is funded with acctReserve + XRP(1). 1 XRP is more than enough for the transaction fees needed in this unit test. So we only need to worry about the incremental reserve that are taken up. Note that in unit test, the incremental reserve for an object is 50XRP, while the transaction fee is only 0.00001XRP

Reply:
ah okay, I missed the fact that Bob has 1 XRP funds 👍